pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -415,9 +417,12 @@ func (m *kubeGenericRuntimeManager) getPodContainerStatuses(uid kubetypes.UID, n
 					tMessage = m.readLastStringFromContainerLogs(status.GetLogPath())
 				}
 			}
-			// Use the termination message written by the application is not empty
+			// Enrich the termination message written by the application is not empty
 			if len(tMessage) != 0 {
-				cStatus.Message = tMessage
+				if len(cStatus.Message) != 0 {
+					cStatus.Message += ": "
+				}
+				cStatus.Message += tMessage
```

Comment:
Is it possible for `cStatus.Message` to grow infinitely, or do we reset it each time this function is called?

Reply:
Thanks for the review.

I can confirm that cStatus.Message will **NOT** grow infinitely, given that `getPodContainerStatuses` is **not stateful** for the cStatus.Message (and it is by design should not be stateful), so that cStatus.Message will **NOT** accumulated.

We can prove this in details, please check:

1. The background is that cStatus.Message will only be updated in 2 places:
![image](https://user-images.githubusercontent.com/32826762/64749175-3b8cfa80-d547-11e9-80c3-f7d35b7faaa3.png)

   Note the cStatus.Message is just `string` instead of `*string`
 https://github.com/kubernetes/kubernetes/blob/f82be3d3d05f67d0d1301ee6bfab4fca0b1a47ac/pkg/kubelet/container/runtime.go#L314

1. cStatus.Message in line 421 can only come from above `toKubeContainerStatus `in the same `getPodContainerStatuses` func:
https://github.com/kubernetes/kubernetes/blob/f82be3d3d05f67d0d1301ee6bfab4fca0b1a47ac/pkg/kubelet/kuberuntime/kuberuntime_container.go#L401

2. In the `getPodContainerStatuses`, it will only **init** the cStatus.Message from `runtime container message`:
https://github.com/kubernetes/kubernetes/blob/f82be3d3d05f67d0d1301ee6bfab4fca0b1a47ac/pkg/kubelet/kuberuntime/kuberuntime_container.go#L459

3. The `runtime container message` can only come directly from the underlay runtime:
https://github.com/kubernetes/kubernetes/blob/f82be3d3d05f67d0d1301ee6bfab4fca0b1a47ac/pkg/kubelet/dockershim/docker_container.go#L400

4. The `runtime container message` will NOT from the cStatus.Message. So, the cStatus.Message will **only be appended at most once** by:
https://github.com/kubernetes/kubernetes/blob/f82be3d3d05f67d0d1301ee6bfab4fca0b1a47ac/pkg/kubelet/kuberuntime/kuberuntime_container.go#L421-L426

5. So the cStatus.Message will **NOT** grow infinitely.

Reply:
Kindly ping @tallclair , PTAL

Reply:
Any comments for this? :)